example/spec/helpers/kinesisHelpers.js

```diff
@@ -30,20 +30,28 @@ const waitPeriodMs = 1000;
  * Helper to simplify common setup code.  wraps function in try catch block
  * that will exit tests if the initial setup conditions fail.
  *
- * @param {Function} fn - function to execute
+ * @param {functino} cleanupCallback - Function to execute if passed in function fails
+ * @param {Function} wrappedFunction - function to execute
  * @param {iterable} args - arguments to pass to the function.
  * @returns {null} - no return
  */
-function tryCatchExit(fn, ...args) {
+async function tryCatchExit(cleanupCallback, wrappedFunction, ...args) {
   try {
-    return fn.apply(this, args);
+    return await wrappedFunction.apply(this, args);
   }
   catch (error) {
-    console.log(error);
-    console.log('Tests conditions can\'t get met...exiting.');
+    console.log(`${error}`);
+    console.log("Tests conditions can't get met...exiting.");
+    try {
+      await cleanupCallback();
+    }
+    catch (e) {
+      console.log(`Cleanup failed, ${e}.   Stack may need to be manually cleaned up.`);
+    }
+    // We should find a better way to do this
```

Comment:
should we be doing it at all?  It seems ok to fail these fast since they are so slow.  But I don't know of another way to do it that is better.